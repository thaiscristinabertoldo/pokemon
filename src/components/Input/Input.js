import React, { memo } from 'react'

import InputStyled from './input.css'

export const Input = memo((props) => {
  return <InputStyled {...props} />
})
