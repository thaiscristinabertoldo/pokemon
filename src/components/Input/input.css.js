import styled from 'styled-components'

import { COLORS } from '../../constants'

const InputStyled = styled.input`
  border-radius: 10px;
  padding: 10px;
  border: solid 1px ${COLORS.lightGrey};
  color: ${COLORS.darkGrey};
  width: 90%;
  &:focus {
    outline: none;
    border: solid 1px ${COLORS.mediumGrey};
  }
`

export default InputStyled
