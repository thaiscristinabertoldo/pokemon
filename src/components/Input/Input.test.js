import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Input } from './Input'

afterEach(cleanup)

describe('Input', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Input />)

    expect(asFragment(<Input />)).toMatchSnapshot()
  })

  it('should take a placeholder', () => {
    const { getByPlaceholderText } = render(<Input placeholder="Placeholder text" />)

    expect(getByPlaceholderText('Placeholder text')).toBeTruthy()
  })
})
