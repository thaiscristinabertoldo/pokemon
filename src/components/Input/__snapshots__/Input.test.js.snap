// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should take a snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm gFuQfl"
  />
</DocumentFragment>
`;
